project/src/main/java/com/google/autograder/data/Database.java

```diff
@@ -254,76 +252,74 @@ public static String getAssignmentSubmissionsData(String courseID, String assign
-      return questionEntity;
-  }
-  
-  
-  // sample add Answer, will change in future
-  public static void addAnswer(String filePath, String parsedAnswer, int score, String assignmentKey, String questionKey) {
-      Entity answerEntity = new Entity("Answer");
-      answerEntity.setProperty("parsedAnswer", parsedAnswer);
-      answerEntity.setProperty("score", score);
-      answerEntity.setProperty("filePath", filePath);
-      answerEntity.setProperty("questionKey", questionKey);
-      answerEntity.setProperty("assignmentKey", assignmentKey);
-      save(answerEntity);
-  }
-  
-  public static Entity addGroup(int score, String questionKey) {
-      Entity groupEntity = new Entity("Group");
-      groupEntity.setProperty("score", score);
-      groupEntity.setProperty("questionKey", questionKey);
-      save(groupEntity);
-      return groupEntity;
-  }
-  
-  //unused right now
-  public static void updateGroupForAnswer(Entity answerEntity, Entity groupEntity) {
-      answerEntity.setProperty("groupKey", groupEntity.getKey());
-      save(answerEntity);
-  }
-  
-  //unused right now
-  public static void updateScoreForAnswer(Entity groupEntity, int score) {
-      groupEntity.setProperty("score", score);
-      save(groupEntity);
-  }
-  
-  public static String getAllQuestionsJSON(String key) {
-    // query all questions with this key at assignment_id key
-    try {
-        Filter propertyFilter = new FilterPredicate("assignmentKey", FilterOperator.EQUAL, key);
-        Query query = new Query("Question").setFilter(propertyFilter);
-        List<Question> questions = new ArrayList<>();
-        for (Entity entity : query(query)) {
-            String name = (String) entity.getProperty("name");
-            Long pointsLong = (Long) entity.getProperty("points");
-            int points = Math.toIntExact(pointsLong);
-            String status = (String) entity.getProperty("status");
-            String assignmentKey = (String) entity.getProperty("assignmentKey");
-            Key questionKey = entity.getKey();
-            Question currQuestion = new Question(name, points, status, assignmentKey, questionKey);
-            questions.add(currQuestion); }
-        String json = new Gson().toJson(questions);
-        System.out.println(json);
-        return json;   
-        
+    
+    public static Entity addQuestion(String questionName, String questionType, int questionPoints, String assignmentKey) {
+        Entity questionEntity = new Entity("Question");
+        questionEntity.setProperty("name", questionName);
+        questionEntity.setProperty("type", questionType);
+        questionEntity.setProperty("points", questionPoints);
+        questionEntity.setProperty("assignmentKey", assignmentKey);
+        save(questionEntity);
+        return questionEntity;
     }
-    catch (Exception e) {
-        System.out.println("assignment wasn't found");
-        return "error";
+    
+    // sample add Answer, will change in future
```

Comment:
These should be javadoc:
```
/**
 * ...
 */
```
and describe what the method does. In general, try to add documentation in the form of javadoc for all public classes and methods.

Reply:
+1 We're going to do this in a separate PR for comments, just so that the creator of each method can properly record the functionality of the methods.